Automattic/WP-Job-Manager
# Add endpoint to get job data with Site Trust verification

Part of #2439 
Based on #2481 

### Changes proposed in this Pull Request

- Implement a REST endpoint which validates the token+user_id+job_id combination and only returns the job data if everything is OK and verified

### Testing instructions

Using this branch of the plugin:

1. Register a job on WPJM
2. On the admin page listing all the jobs (`/wp-admin/edit.php?post_type=job_listing`), click on the "Promote" button
3. On the modal that appears, click on the "Promote your jobs" button
4. Check that the plugin validates that you're using HTTPS (if you're not, feel free to comment the condition with `is_ssl()` for checking)
5. Make sure you're directed to a page like `https://wpjobmanager.com/promote-job?user_id=YOUR_USER_ID&job_id=YOUR_JOB_ID&token=A_RANDOM_TOKEN&site_url=THE_HTTPS_VERSION_OF_YOUR_SITE`
6. Get the Job ID,  user ID, token and site URL specified and make sure you can access `http://YOUR_SITE/wp-json/wpjm-internal/v1/promoted-jobs/YOUR_JOB_ID?user_id=YOUR_USER_ID&token=A_RANDOM_TOKEN`, and that it returns `verified` set to `true` with all the job data of the specified Job ID in the `job_data` key
7. Also make sure that, if you make the same request again, it will return `verified` set to `false`, and no job data

Also, make sure that the API Doc for that endpoint is correct
